### PR TITLE
[docs] Fix google-sign-in example & GoogleService-Info.plist updates

### PR DESCRIPTION
**Pull request by IjzerenHein**
# Why

- fixes several imports in example code which no longer worked
- renames `GoogleService-info.plist` -> `GoogleService-Info.plist`
- update ios workflow to use GoogleService-Info plist by default
- added separate section on optional firebase configuration

# How

Tested examples in test-app

# Test Plan

- on iOS, add GoogleService-Info.plist to app.json
- omit `clientId` from `initAsync` method on iOS
- sign-in should still work

### CODE REVIEW

**Comment by raRaRa** [comment]
This is likely unrelated, but why do we need `GoogleService-Info.plist` and `google-services.json`?

1. I don't want to use Firebase
2. I want to specify the properties myself in app.json, e.g. set the `reservedClientId` myself for iOS and Android.

This causes a lot of confusion for users who don't want to use Firebase. I already have Android and iOS OAuth clients in my Google dev console, how would I go about adding them manually?

**Comment by IjzerenHein** [comment]
> This is likely unrelated, but why do we need GoogleService-Info.plist and google-services.json?

@EvanBacon do you know whether this is needed?

**Comment by raRaRa** [comment]
Shouldn't `"reservedClientId": "<YOUR_IOS_CLIENT_ID>"` be `"reservedClientId": "<YOUR_IOS_REVERSED_CLIENT_ID>"`?

File: docs/pages/versions/unversioned/sdk/google-sign-in.md [hunk 1]
```diff
@@ -24,9 +24,11 @@ For questions on setup, feel free to comment on this post: [**React Native Googl
...
 2. Open up [the Firebase Console](https://console.firebase.google.com) and setup a new project, or use an existing one.
 3. Create a native iOS, and Android app using the Bundle ID and Android package you defined earlier.
-4. Download the `GoogleService-info.plist` (iOS) & the `google-services.json` (Android). Move them to your Expo project.
-5. In the `app.json`, set your `expo.ios.config.googleSignIn.reservedClientId` to the value of `REVERSE_CLIENT_ID` in the `GoogleService-info.plist`.
-6. Also in `app.json`, set `expo.android.googleServicesFile` to the relative path of your `google-services.json`. Make sure the file is located somewhere in your Expo project.
+4. Download the `GoogleService-Info.plist` (iOS) & the `google-services.json` (Android). Move them to your Expo project.
+5. In `app.json`, set your `expo.ios.config.googleSignIn.reservedClientId` to the value of `REVERSE_CLIENT_ID` in the `GoogleService-Info.plist`.
+6. Also in `app.json`, set `expo.ios.googleServicesFile` to the relative path of your `GoogleService-Info.plist`. Make sure the file is located somewhere in your Expo project.
```
Comment (EvanBacon) [inline docs/pages/versions/unversioned/sdk/google-sign-in.md]: 😍

**Review by EvanBacon — APPROVED** [review]
Awesome!

**Comment by IjzerenHein** [comment]
> Shouldn't "reservedClientId": "<YOUR_IOS_CLIENT_ID>" be "reservedClientId": "<YOUR_IOS_REVERSED_CLIENT_ID>"?

@raRaRa Yes it should be, I've updated it in this PR 👍

**Comment by raRaRa** [comment]
A few more things to fix:

Under `GoogleUser` variables:
`auth: ?Authentication;` should probably be `auth: ?GoogleAuthentication;`

`Extends GoogleIdentity, manaages all data regarding an authenticated user.` has a typo, maanages should be manages.

Thanks!

**Comment by IjzerenHein** [comment]
> A few more things to fix:
> 
> Under `GoogleUser` variables:
> `auth: ?Authentication;` should probably be `auth: ?GoogleAuthentication;`
> 
> `Extends GoogleIdentity, manaages all data regarding an authenticated user.` has a typo, maanages should be manages.
> 
> Thanks!

Thanks, I've updated the PR!

**Comment by raRaRa** [comment]
Not sure if it's related to this PR, but the master branch has some issues: https://github.com/expo/expo/blob/master/docs/pages/versions/unversioned/sdk/google-sign-in.md

It looks like the `Methods` section isn't being rendered correctly.

**Comment by IjzerenHein** [comment]
> Not sure if it's related to this PR, but the master branch has some issues: https://github.com/expo/expo/blob/master/docs/pages/versions/unversioned/sdk/google-sign-in.md
> 
> It looks like the `Methods` section isn't being rendered correctly.

No that's just fine and gets replaced & rendered on the Expo website.

File: docs/pages/versions/unversioned/sdk/google-sign-in.md [hunk 2]
```diff
@@ -37,16 +45,18 @@ For questions on setup, feel free to comment on this post: [**React Native Googl
       "bundleIdentifier": "example.expo.googlesignin",
       "config": {
         "googleSignIn": {
-          // Your REVERSE_CLIENT_ID from the GoogleService-info.plist
-          "reservedClientId": "<YOUR_IOS_CLIENT_ID>"
+          // Your REVERSED_CLIENT_ID from the GoogleService-Info.plist
```
Comment (seigel) [inline docs/pages/versions/unversioned/sdk/google-sign-in.md:48]: reSERVed or reVERSed?

Comment (brentvatne) [reply docs/pages/versions/unversioned/sdk/google-sign-in.md:48]: reserved. it's a particularly bad and entrenched typo.

https://github.com/expo/expo/pull/10274#issuecomment-696504888

**Pull request merged**